python/cudf/cudf/utils/cudautils.py

```diff
@@ -195,7 +195,7 @@ def find_first(arr, val, compare="eq"):
         return min_index
 
 
-def find_last(arr, val, compare="eq"):
+def find_last(arr, val, compare="eq", sentinel_value=-1):
```

Comment:
Adding a sentinel value in here is really unsafe, especially for types like `int8` or `uint8`. Given this kernel is read_only, instead of passing just the data buffer, we could pass both the data and mask buffer.

If the mask buffer is None we dispatch to a non-null kernel, otherwise we can check the mask bit as needed to find the value.